tsl/src/hypercore/utils.c

```diff
@@ -54,6 +57,21 @@ hypercore_set_am(const RangeVar *rv)
 		};
 
 		recordDependencyOn(&depender, &referenced, DEPENDENCY_NORMAL);
+
+		/*
+		 * Update the tuple for the compressed chunk and disable autovacuum on
+		 * it. This requires locking the relation (to prevent changes to the
+		 * definition), but it is sufficient to take an access share lock.
+		 */
+		Chunk *chunk = ts_chunk_get_by_relid(relid, true);
```

Comment:
Can we do the autovacuum settings outside this function to keep it clean in such a way that it only does changes to the access method used. Technically,this code could be called on any rel previously, but now it requires it to be a chunk. While we don't yet support setting Hypercore TAM on non-chunk tables, I think we should avoid having "chunks" creep into all our code.

You could wrap this function in another one that sets the AM and then updates vacuum settings.

Reply:
Besides, there's already a chunk object outside this function so it makes more sense to move it out to avoid building the chunk again.